services/agents/activities/utils.ts

```diff
@@ -0,0 +1,81 @@
+export function getMonthlyBillingCycle(
```

Comment:
Question:

Is this method guaranteed to work regardless of the timezone in which it is executed?

Reply:
Thank you. The answer is "No".

- Given `referenceDate` is from the database which is stored in UTC. So this may be okay.
- Given `curerntDate`, I passed `new Date()` and this is wrong. This must be fixed.

I think using a good library like `date-fns` or calculating all values as UTC is better to ensure accurate value.
I'll change this.

----

And timestamp value in the database must be fixed.
ref https://github.com/giselles-ai/giselle/issues/39

Reply:
> Given referenceDate is from the database which is stored in UTC. So this may be okay.

Sorry, ⬆️ This my understanding was wrong, also.

```ts
// in JST
const d = new Date("2024-07-31T19:17:10.156Z")
d.getDate() // => 1
d.getUTCDate() // => 31
```

I thought `d.getDate()` should return `31` but it does not.
We must use the `getUTCXXX` version.

Reply:
I've fixed timezone issue with additional test cases.